pxtcompiler/emitter/service.ts

```diff
@@ -927,7 +927,9 @@ namespace ts.pxtc.service {
                     /^__/.test(si.name) || // ignore members starting with __
                     /^__/.test(si.namespace) || // ignore namespaces starting with _-
                     si.attributes.hidden ||
-                    si.attributes.deprecated
+                    si.attributes.deprecated ||
+                    // don't emit members of enum marked as "emitAsConstant"
+                    (si.kind == SymbolKind.EnumMember && lastApiInfo.apis.byQName[si.namespace]?.attributes.emitAsConstant)
```

Comment:
Will this work if the enum is declared inside of a namespace? Wilt the namespace field have the full qualified name?

Reply:
i'm going to rely on an annotation rather than weird logic